pkg/api/v1/types.go

```diff
@@ -1475,6 +1475,28 @@ type PodLogOptions struct {
 	Previous bool `json:"previous,omitempty" description:"return previous terminated container logs; defaults to false"`
 }
 
+// PodAttachOptions is the query options to a Pod's remote attach call
+// TODO: merge w/ PodExecOptions below for stdin, stdout, etc
+type PodAttachOptions struct {
+	TypeMeta `json:",inline"`
+
+	// Stdin if true indicates that stdin is to be redirected for the attach call
+	Stdin bool `json:"stdin,omitempty" description:"redirect the standard input stream of the pod for this call; defaults to false"`
+
+	// Stdout if true indicates that stdout is to be redirected for the attach call
+	Stdout bool `json:"stdout,omitempty" description:"redirect the standard output stream of the pod for this call; defaults to true"`
```

Comment:
How is it possible to default these values to true? We can't distinguish the case they are set to false by the user and set to false by Go. They'd all need to be pointer fields to do that.